src/activities/assignments/AssignmentEntity.js

```diff
@@ -554,6 +563,30 @@ export class AssignmentEntity extends Entity {
 		await performSirenAction(this._token, action, fields);
 	}
 
+	/**
+	 * Sets the custom allowable filetypes of the assignment
+	 * @param {string} customAllowableFileTypes Allowable filetype option
+	 */
+	async setCustomAllowableFileTypes(customAllowableFileTypes) {
+		customAllowableFileTypes = String(customAllowableFileTypes);
+		const action = this.canEditAllowableFileType() && this._entity.getActionByName(Actions.assignments.updateCustomAllowableFileType);
+		if (!action) {
+			return;
+		}
+
+		const fieldValue = action.getFieldByName('customAllowableFileTypes').value;
+
+		if (!fieldValue) {
+			return;
+		}
+
+		const fields = [
+			{ name: 'customAllowableFileTypes', value: customAllowableFileTypes }
```

Comment:
Could `customAllowableFileTypes` be undefined, and if it is, do we still want to continue performing the siren action?

Reply:
Theres a check to see if it is undefined. It does not continue if it is.